### ISSUE TITLE
[Android] No plans - IAP screen displayed after a cancelled Mozilla VPN subscription expires

### ISSUE THREAD

**Issue by data-sync-user** [mozilla-mobile/mozilla-vpn-client#6260]
**Important details:**

* Not being able to reproduce on iOS;
* FxA does not list the subscription for that account;

**Mozilla VPN affected versions:**

* 2.14.0 (2.202303021512);

**Platform tested:**

* Android;

**Prerequisites:**

* Have the Mozilla VPN  installed;
* Have an account with subscription for Mozilla VPN;
* Be signed in;
* Have the VPN turned ON;

**Steps to reproduce:**

1. Go to Play Store - Subscriptions and cancel the subscription for Mozilla VPN;
1. Leave the VPN ON (while being logged with the account with subscription canceled);
1. Wait for 24h so that the subscription expires;
1. Observe the behavior;

**Expected results:**

* User is logged out / the IAP screen is displayed.

**Actual results:**

* User is prompted with the IAP screen with **no plans**.

**Notes:**

* Signing back in will redirect the user to the correct IAP screen.

┆Issue is synchronized with this [Jira Bug](https://mozilla-hub.atlassian.net/browse/VPN-4318)
┆Reporter: Valentina Virlics

**Comment by data-sync-user** [comment]
➤ Andrea Marchesini commented:

The latest log file contains the entire sequence of steps. It seems a bug in PurchaseHandler.

**Comment by data-sync-user** [comment]
➤ Valentina Virlics commented:

Adrienne Davenport We can confirm that this is happening on 2.13 as well.

Andrea Marchesini Attaching another set of logs, from another Android device.

[^MozillaVPN_Logs_2023-18-11-10-18-21.txt]

**Comment by data-sync-user** [comment]
➤ Andrea Marchesini commented:

Sarah Bird can you please take a look at the log? It seems a PurchaseHolder bug.

**Comment by data-sync-user** [comment]
➤ Betty Fleming commented:

Reproducible in 2.13, and happens only after expiration. Santiago okay with not blocking the 2.14 release.

**Comment by data-sync-user** [comment]
➤ Valentina Virlics commented:

We made the setup to be able to verify this ticket. We used 2.15.0 (2.202305041546) on Android 10.

Will update the ticket after testing is complete.

**Comment by data-sync-user** [comment]
➤ Valentina Virlics commented:

We finalized in verifying this on Android 10/11, using Mozilla VPN 2.15.0 (2.202305041546) build.

After 24h, the IAP screen was triggered:

  * on the Android 10 device - IAP screen and a spinning wheel instead of plans (see video below);
  * on the Android 11 device - IAP screen + spinning wheel (only for few seconds) and after that the plans were displayed;

The OS notification ( https://mozilla-hub.atlassian.net/browse/VPN-3434 ) indicated as if the Mozilla VPN was still up and running.

Not sure why we see 2 different behaviors. Attaching logs from the device where the plans did not load after few seconds.

[^MozillaVPN_Logs_2023-08-07-14-8-13.txt]

!2023_05_07_14_06_30.mp4|width=360,height=640!

Basti  Will recheck again, on the device where the plans were not displayed, to be able o confirm or not the behavior. Will reopen the ticket if this behavior confirms.

**Comment by data-sync-user** [comment]
➤ Valentina Virlics commented:

We finalized the retesting on the device we saw for a few minutes the spinning wheel instead of the plans, and we were not able to reproduce it again.

After 24h from the subscription cancellation, the user was redirected to IAP screen and the plans were correctly displayed,

We will continue motorizing this, and track separately future issues if the case.

Marking this as verified fixed.
